spec/extensions/fourier_transform_functions.md

```diff
@@ -0,0 +1,505 @@
+
+Computes the one-dimensional inverse discrete Fourier transform for real-valued input.
+
+#### Parameters
+
+-   **a**: _&lt;array&gt;_
+
+    -   Input array.
+
+-   **n**: _int_
+
+    -   Length of the transformed axis of the output. If given, the input will be either zero-padded or trimmed to this length before computing the real Fourier transform. Otherwise, it will default to `2 * (m - 1)` where `m` is the length of the input along the axis given by the `axis` keyword. Default: `None`.
+
+-   **axis**: _int_
+
+    -  Axis used to compute the real Fourier transform. If it is not specified, the last axis is used. Default: `-1`.
+
+-   **norm**: _str_
+
+    -   Specify the normalization mode. Should be one of the following modes:
+
+        - `'backward'`: Normalize by `1/n`.
+        - `'ortho'`: Normalize by `1/sqrt(n)`
+        - `'forward'`: No normalization.
+
+        Default: `'backward'`
+
+#### Returns
+
+-   **out**: _&lt;array&gt;_
+
+    -   An array transformed along the axis indicated by the `axis` keyword.
+
+#### Raises
+
+-   If `axis` is not a valid axis of `a`.
+
+(function-rfftn)=
+### rfftn(a, /, *, s=None, axes=None, norm='backward')
+
+Computes the n-dimensional discrete Fourier transform for real-valued input.
+
+#### Parameters
+
+-   **a**: _&lt;array&gt;_
+
+    -   Input array.
+
+-   **s**: _Union\[ Sequence\[ int ], Tuple\[ int, ... ] ]_
+
+    -   Size of each transformed axis of the output. If given, each axis will be either zero-padded or trimmed to the length `s[i]` before computing the real Fourier transform. Otherwise, no padding will be performed in each dimension. Default: `None`.
+
+-   **axes**: _Union\[ Sequence\[ int ], Tuple\[ int, ... ] ]_
+
+    -  Axes over which to compute the Fourier transform. If not specified, the last `len(s)` axes are used, or all axes if `s` is not specified either. Default: `None`.
+
+-   **norm**: _str_
+
+    -   Specify the normalization mode. Should be one of the following modes:
+
+        - `'backward'`: No normalization.
+        - `'ortho'`: Normalize by `1/sqrt(n)`
+        - `'forward'`: Normalize by `1/n`.
+
+        Default: `'backward'`
+
+#### Returns
+
+-   **out**: _&lt;array&gt;_
+
+    -   An array transformed along the axes indicated by the `axes` keyword.
+
+#### Raises
+
+-   If `s` and `axes` have different lengths.
+-   If an element of `axes` is larger than the number of axes of `a`.
+
+(function-irfftn)=
+### irfftn(a, /, *, s=None, axes=None, norm='backward')
+
+Computes the n-dimensional inverse discrete Fourier transform for real-valued input.
+
+#### Parameters
+
+-   **a**: _&lt;array&gt;_
+
+    -   Input array.
+
+-   **s**: _Union\[ Sequence\[ int ], Tuple\[ int, ... ] ]_
+
+    -   Size of each transformed axis of the output. If given, each axis will be either zero-padded or trimmed to the length `s[i]` before computing the Fourier transform. Otherwise, no padding will be performed in each dimension. Default: `None`.
```

Comment:
```suggestion
    -   Size of each transformed axis of the *output*. If given, the last axis will be either zero-padded or trimmed to `s[-1]//2+1`, whereas all other axes `i` are either zero-padded or trimmed to the length `s[i]`, before computing the inverse of `rfftn`. Otherwise, the last axis is either zero-padded or trimmed to `2 * (m - 1)`, where `m` is the length of the input along the axis, and all other axes use the input shape. The last element `s[-1]` is for computing `irfft(a[axes[-1]], n=s[-1])` whereas other elements for `ifft(a[axes[i]], n=s[i])`. Default: `None`.
```